### DISCUSSION TITLE
How to Create dfs1 files?

### DISCUSSION THREAD

**Discussion by mecwes** [DHI/mikeio#572] [Q&A] [answered]
Hi,
I'd like to create dfs1 files with boundary conditions such as wave parameters, water levels, currents, etc. to be used in MIKE21-SW, MIKE21-HD and MIKE3 model runs. 
I'm able to find the boundary locations when reading a mesh file and do interpolations to these locations. The next step is to create various data-arrays (Hs, Tp, Dir, etc) with the right dimensions (point, time), the data interpolated to the mesh boundary locations and then save these to a dfs1 file. 
How can I create a dfs1 file with the correctly constructed datasets, dataarrays, grids, timestamps, parameter names, units, etc. such that they are accepted by MIKE21? If some guidance on how to make a dfs1 file from scratch can be given would help a lot.
Hope to hear.

Kind regards,
Marco Westra

**Comment by ecomodeller** [comment, marked as answer]
I think you can get guidance on the correct EUM Types that can be used in a MIKE 21 SW when you are setting up the model in MIKE Zero.

For some help on creating a dfs1 file take a look at https://dhi.github.io/getting-started-with-mikeio/dfs1.html

**Reply by mecwes** [reply]
Thanks, in the end, I was able to make the dfs1 files I wanted. However, I'm still puzzled how MIKE21 knows the right lat, lon positions in the dfs1 file or how to retrieve them, as they are not at regular distances.
Rgds.

**Reply by ecomodeller** [reply]
> Thanks, in the end, I was able to make the dfs1 files I wanted. However, I'm still puzzled how MIKE21 knows the right lat, lon positions in the dfs1 file or how to retrieve them, as they are not at regular distances.
> Rgds.

I am pretty sure the engine doesn't use any spatial information from the dfs1 file, it is applied along the specified boundary at equally spaced intervals.

**Reply by jsmariegaard** [reply]
I agree, the MIKE 21 engine does not use the spatial information in the dfs1 file but considers it just as equidistant values on a line (CCW direction). It then interpolates to the relevant positions using a relative coordinate s from start (0) to end (1).
